clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -318,4 +321,16 @@ public AttributeList setAttributes(AttributeList list) {
                                       + ".(whitelist/blacklist) is not a valid regular expression");
         }
     }
+
+    @Override
+    public void contextChange(MetricsContext metricsContext) {
+        if (!this.mbeans.isEmpty()) {
```

Comment:
accesss to `mbeans` needs to be guarded by the lock as well, since the map is not thread-safe.